econml/metalearners.py

```diff
@@ -262,6 +262,8 @@ def fit(self, Y, T, X, *, inference=None):
         """
         # Check inputs
         Y, T, X, _ = check_inputs(Y, T, X, multi_output_T=False)
+        if Y.ndim == 2 and Y.shape[1] == 1:
+            Y = Y.flatten()
```

Comment:
Why is this necessary?  Please add a comment explaining, and a test that covers this scenario.

Reply:
The `sklearn` models flatten the output and the difference between an `(n, 1)` and an `(n, )`  is an `(n, n)` array. Since there are several models where we compute differences this way and it's difficult to keep track of the shapes, it's more straightforward to just flatten the input `Y`, I'll make sure to add a comment that reflects this.